operator/pkg/controllers/controller.go

```diff
@@ -42,90 +45,53 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	// 1. Read Spec
 	resource := c.For()
 	if err := c.Get(ctx, req.NamespacedName, resource); err != nil {
-		if errors.KubeObjNotFound(err) {
+		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
-		return reconcile.Result{}, err
+		return *status.Failed, err
 	}
 	// 2. Copy object for merge patch base
 	persisted := resource.DeepCopyObject()
-	// 3. reconcile else finalize if object is deleted
-	result, err := c.reconcile(ctx, resource)
-	if err != nil {
-		resource.StatusConditions().MarkFalse(v1alpha1.Active, "", err.Error())
-		if errors.SafeToIgnore(err) {
-			return result, nil
-		}
-		zap.S().Errorf("Failed to reconcile kind %s, %v", resource.GetObjectKind().GroupVersionKind().Kind, err)
-		return reconcile.Result{Requeue: true}, err
+	// 3. Reconcile else finalize if object is deleted
+	result, reconcileErr := c.reconcile(ctx, resource, persisted)
+	// 4. Update Status using a merge patch, we want to set status even when reconcile errored
+	if err := c.Status().Patch(ctx, resource, client.MergeFrom(persisted)); err != nil && !errors.IsNotFound(err) {
+		return *status.Failed, fmt.Errorf("status patch for %s, %w,", req.NamespacedName, err)
 	}
-	resource.StatusConditions().MarkTrue(v1alpha1.Active)
-	// 4. Update Status using a merge patch
-	if err := c.Status().Patch(ctx, resource, client.MergeFrom(persisted)); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to persist changes to %s, %w", req.NamespacedName, err)
+	if reconcileErr != nil {
+		return *status.Failed, reconcileErr
 	}
 	return result, nil
 }
 
-func (c *GenericController) reconcile(ctx context.Context, resource Object) (reconcile.Result, error) {
+func (c *GenericController) reconcile(ctx context.Context, resource Object, persisted runtime.Object) (reconcile.Result, error) {
+	var result *reconcile.Result
+	var err error
+	existingFinalizers := resource.GetFinalizers()
```

Comment:
Since this isnt used elsewhere, consider 

`existingFinalizers := sets.NewString(resource.GetFinalizers())`